src/main/java/mods/eln/misc/DirectionSet.kt

```diff
@@ -0,0 +1,46 @@
+package mods.eln.misc
+
+import net.minecraft.nbt.NBTTagCompound
+import java.io.DataInputStream
+import java.io.DataOutputStream
+import java.util.*
+
+class DirectionSet: TreeSet<Direction>() {
+    fun toInt(): Int {
+        var value = 0
+        if (contains(Direction.XN)) value += 1
```

Comment:
Kotlin does not have |= and << operators afaik.

I would have to do this: 
value = value or 1.shl(x)

Not very readable imho. Maybe I find something smarter than that...